src/opm/parser/eclipse/EclipseState/Schedule/Well/WellTestState.cpp

```diff
@@ -161,6 +172,32 @@ namespace Opm {
         return well_iter->last_test;
     }
 
+    void WellTestState::updateForNewWTEST(const Opm::WellTestConfig& config)
+    {
+        // check whether to reset based on the new WTEST request
+        for (auto& well: this->wells) {
+            if (config.has(well.name, well.reason)) {
+                const auto& well_config = config.get(well.name, well.reason);
+                if (well_config.begin_report_step > well.wtest_report_step) {
+                    // there is a new WTEST input, we should reset the counting
+                    well.num_attempt = 0;
+                    well.wtest_report_step = well_config.begin_report_step;
+                }
+                assert(well_config.begin_report_step == well.wtest_report_step);
```

Comment:
OK - please use and exception instead:
```
if (well_config.begin_report_step != well.wtest_report_step)
     throw std::runtime_error("Bug in opm/flow - .... ");
```

Reply:
Sure. I will do it.

Reply:
If it is a programming error that would cause the exception to trigger (I am assuming it is, since it will replace an assert()) it should be `logic_error` not `runtime_error`.

Reply:
I prefer an assert() since it is not something that users cause.  okay. a throw will happen here, it will work similarly.

Reply:
One question, how `assert` and `throw` work differently in parallel case?

Two processes, one process `assertion failure`. 

vs 

Two processes, one process `throw runtime_error`.

Reply:
Both will be bad, and result in MPI killing the other processes. But assert() having that effect is sort-of expected. With a throw there is a possibility to catch and communicate, but it may not be worth it, this is expected to crash and burn to point out a bug.

Reply:
> but it may not be worth it, this is expected to crash and burn to point out a bug.

As long as they will not create deadlock, it is okay to me. I know throw can create deadlock, while I am not sure about `assert`.